app/slave/subjob_executor.py

```diff
@@ -131,14 +131,15 @@ def _execute_atom_command(self, atomic_command, atom_environment_vars, atom_arti
         :rtype: int
         """
         fs_util.create_dir(atom_artifact_dir)
+        # This console_output_file must be opened in 'w+b' mode in order to be interchangeable with the
+        # TemporaryFile instance that gets instantiated in self._project_type.execute_command_in_project.
+        console_output_file = open(os.path.join(atom_artifact_dir, Subjob.OUTPUT_FILE), mode='w+b')
```

Comment:
One thing that popped into my head here was that we need to make sure this gets closed. I can't tell if the `TemporaryFile` objects we were using before do more to make sure they get closed than a simple `open()`, but it might be more important since it's not going to be as big of a deal if we're holding open a random tempfile.

Also, I see in the happy path case we should close the file in `ProjectType._read_file_contents_and_close()` but if we hit an exception before that we'd miss that close call (it's not in a finally block).

I wonder if we should just use the `with open(...)` syntax here around the call to `self._project_type.execute_command_in_project` just to make sure it gets closed?